ui/js/directives/treeherder/clonejobs.js

```diff
@@ -249,32 +256,32 @@ treeherder.directive('thCloneJobs', [
             jobStatus.key = getJobMapKey(job);
             jobStatus.value = job.job_type_symbol;
             jobStatus.title = getHoverText(job);
-
+            jobStatus.extraClasses = "";
+            jobStatus.extraClasses += job.visible ? " filter-shown" : "";
+            if( !_.isEmpty(lastJobSelected.job) &&
+                (lastJobSelected.job.id === job.id)){
+                jobStatus.extraClasses += " " + largeBtnCls + " " + selectedBtnCls;
+            } else {
+                jobStatus.extraClasses += " " + btnCls;
+            }
             if (thResultStatus(job) === "runnable") {
                 jobStatus.buildername = job.ref_data_name;
-                jobBtn = $(runnableJobBtnInterpolator(jobStatus));
-
                 if (ThResultSetStore.isRunnableJobSelected($rootScope.repoName,
                                                            job.result_set_id,
                                                            jobStatus.buildername)) {
-                    jobBtn.addClass("runnable-job-btn-selected");
+                    jobStatus.extraClasses += " runnable-job-btn-selected";
                 }
+                jobBtn = runnableJobBtnInterpolator(jobStatus);
             } else {
-                jobBtn = $(jobBtnInterpolator(jobStatus));
+                jobBtn = jobBtnInterpolator(jobStatus);
             }
-
             // If the job is currently selected make sure to re-apply
             // the job selection styles
             if( !_.isEmpty(lastJobSelected.job) &&
                 (lastJobSelected.job.id === job.id)){
-
-                setSelectJobStyles(jobBtn);
```

Comment:
@wlach, how can we set the styles when the element has not been created? we are just concatenating HTML at this stage right?